src/store/page-load/saga.ts

```diff
@@ -0,0 +1,30 @@
+import { getHistory } from '../../lib/browser';
+import { featureFlags } from '../../lib/feature-flags';
+import { getCurrentUserWithChatAccessToken } from '../authentication/saga';
+import { initializePublicLayout } from '../layout/saga';
+
+const anonymousPaths = [
+  '/get-access',
+  '/login',
+];
+
+export function* saga() {
+  const history = getHistory();
+  if (anonymousPaths.includes(history.location.pathname)) {
+    return;
+  }
+
+  const success = yield getCurrentUserWithChatAccessToken();
+  if (success) {
+    return;
+  }
+
+  if (featureFlags.allowPublicZOS) {
+    yield initializePublicLayout();
+    return;
+  }
+
+  history.replace({
+    pathname: '/login',
+  });
+}
```

Comment:
I actually don't understand this file. We're not listening for a saga event here. Where does this code gets triggered/used? 🤔

Reply:
This code gets called immediately when the global `store/saga.ts` is loaded. It's the first file in the list of sagas that get run.